qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/QuteLanguageServer.java

```diff
@@ -288,4 +295,16 @@ public void setTrace(SetTraceParams params) {
 		// to avoid having error in vscode, the method is implemented
 		// FIXME : implement the behavior of this method.
 	}
+
+	@Override
+	public Template getTemplate(String uri) {
+		TextDocumentIdentifier textDocument = new TextDocumentIdentifier(uri);
+		AbstractTextDocumentService abstractTextDocumentService = textDocumentService.getTextDocumentService(textDocument);
+		if (abstractTextDocumentService.getFileType() == FileType.TEMPLATE) {
+			TemplateFileTextDocumentService templateTextDocumentService = (TemplateFileTextDocumentService) abstractTextDocumentService;
+			QuteTextDocument document = templateTextDocumentService.getDocument(uri);

```

Comment:
This code should be moved in TemplateFileTextDocumentService:

 * TemplateFileTextDocumentService should implements QuteTemplateProvider which is implemented with the following code

```java
@Override
public getTemplate(String uri) {
   QuteTextDocument document = getDocument(uri);
   return document != null ? document.getModel() : null;
}
```
 
After that you can write here:

`return ((TemplateFileTextDocumentService) abstractTextDocumentService).getTemplateUri(uri);`